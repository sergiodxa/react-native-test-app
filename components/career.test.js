import React from 'react';
import Career from './career';

import renderer from 'react-test-renderer';

const courses = [
  { badge: 'https://static.platzi.com/media/achievements/comunidad.png', name: 'Comunidad Platzi' },
  { badge: 'https://static.platzi.com/media/achievements/programacion-basica2x.png', name: 'Curso Gratis de Programación Básica' },
  { badge: 'https://static.platzi.com/media/achievements/badge-fundamentos_1.png', name: 'Fundamentos de Ingeniería de Software' },
  { badge: 'https://static.platzi.com/media/achievements/03-1.png', name: 'Curso de Programación para Padres y Niños' },
];

const career = { title: 'Fundamentos Básicos', courses };

describe('Career item', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<Career {...career} />).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders the expected UI', () => {
    const rendered = renderer.create(<Career {...career} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
