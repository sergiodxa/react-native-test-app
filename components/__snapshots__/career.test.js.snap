// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Career item renders the expected UI 1`] = `
<View
  style={
    Array [
      Object {},
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "fontSize": 24,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Fundamentos Básicos
  </Text>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://static.platzi.com/media/achievements/comunidad.png",
        }
      }
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 8,
            "width": 36,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Comunidad Platzi
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://static.platzi.com/media/achievements/programacion-basica2x.png",
        }
      }
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 8,
            "width": 36,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Curso Gratis de Programación Básica
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://static.platzi.com/media/achievements/badge-fundamentos_1.png",
        }
      }
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 8,
            "width": 36,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Fundamentos de Ingeniería de Software
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderBottomColor": "black",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "display": "flex",
          "flexDirection": "row",
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://static.platzi.com/media/achievements/03-1.png",
        }
      }
      style={
        Array [
          Object {
            "height": 36,
            "marginRight": 8,
            "width": 36,
          },
          undefined,
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Curso de Programación para Padres y Niños
    </Text>
  </View>
</View>
`;
